rules.d/60-block.rules

```diff
@@ -1,8 +1,7 @@
 # do not edit this file, it will be overwritten on update
 
 # enable in-kernel media-presence polling
-ACTION=="add", SUBSYSTEM=="module", KERNEL=="block", ATTR{parameters/events_dfl_poll_msecs}=="0", \
-  ATTR{parameters/events_dfl_poll_msecs}="2000"
+ACTION=="add", ATTR{removable}=="1", ATTR{events_poll_msecs}=="-1", ATTR{events_poll_msecs}="2000"
```

Comment:
To support the non-builtin case, maybe better to keep the original rules? Or is the new rules cover both cases?

Reply:
I believe the original intent of the rule was to set polling to removable devices such as CDROM and DVDROM drives, per Kay's commit c5a41da949.  The commit message doesn't explain the intent, but looking at changes made by Kay at the time https://github.com/mcatalancid/libudev/blob/3d27a9942a882a4d7360b2a5ef28fa0f989288bf/ChangeLog#L340 it shows a couple audio/CDROM items being tackled.

The new rule more correctly handles this use case by using the `removable` attribute instead of the specifics of the event name or subsystem, so it should replace it, in my opinion.

Reply:
Hm, still I think we should keep the current rules. I am not familiar with the attribute, but the rules for the module should have no race (at least if no disc inserted on boot), but the per-drive rules added in this PR sounds to have races potentially, as udevd may process the event slightly later after the kernel emits the event. In the (hopefully short but still finite) timespan, there may be some events that should be polled. But, if the rules are applied slightly later, the event may be lost. No? Or do I misunderstand something?

Reply:
Hmm, of Fedora, the block driver seems to be built-in. But, the attribute is set.
```sh
$ cd /usr/lib/modules/6.13.11-200.fc41.x86_64
$ tree | grep block
│   │   ├── block
│   │   │   │   └── mspro_block.ko.xz
│   │   │   │   ├── mmc_block.ko.xz
│   │   │   │   ├── block2mtd.ko.xz
│   │   │   ├── mtdblock.ko.xz
│   │   │   ├── target_core_iblock.ko.xz
│   │   │   ├── blocklayout
│   │   │   │   └── blocklayoutdriver.ko.xz
├── modules.block
$ cat /sys/module/block/parameters/events_dfl_poll_msecs 
2000
```
I guess, your system does not enable `systemd-udev-trigger.service`, or the command in the service is modified and module devices are not triggered.
Please provide the result of `systemctl status systemd-udev-trigger.service`.

Reply:
Guix System doesn't use systemd as it init system (it uses [Shepherd](https://www.gnu.org/software/shepherd/)), so I cannot run the above command.  Do I understand that for the `events_dfl_poll_msecs` property to be set correctly, a service external to udev is needed?  Is this documented somewhere?

Reply:
Even if you do not use systemd (even if you use eudev), triggering events by `udevadm trigger` is necessary for making the rules applied. Please contact to your distro.